lib/libmp/src/core/mp_structure.c

```diff
@@ -0,0 +1,299 @@
+/*
+ * Copyright 2025 NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <stdio.h>
+
+#include <zephyr/kernel.h>
+
+#include "mp_structure.h"
+#include "mp_value.h"
+
+typedef struct {
+	const char *name;
+	MpValue *value;
+	sys_snode_t node;
+} MpStructureField;
+
+void mp_structure_init(MpStructure *structure, const char *name)
+{
+	if (structure != NULL) {
+		structure->name = name;
+		sys_slist_init(&structure->fields);
+	}
+}
+
+MpStructure *mp_structure_new_empty(const char *name)
+{
+	MpStructure *structure = k_malloc(sizeof(MpStructure));
+
+	mp_structure_init(structure, name);
+
+	return structure;
+}
+
+void mp_structure_clear(MpStructure *structure)
+{
+	MpStructureField *field;
+
+	if (structure == NULL) {
+		return;
+	}
+
+	while (!sys_slist_is_empty(&structure->fields)) {
+		field = CONTAINER_OF(sys_slist_get(&structure->fields), MpStructureField, node);
+		mp_value_destroy(field->value);
+		k_free(field);
+	}
+}
+
+void mp_structure_destroy(MpStructure *structure)
+{
+	mp_structure_clear(structure);
+	k_free(structure);
+}
+
+void mp_structure_append(MpStructure *structure, const char *name, MpValue *value)
+{
+	MpStructureField *field;
+
+	if (structure == NULL || name == NULL || value == NULL) {
+		return;
+	}
+
+	field = k_malloc(sizeof(MpStructureField));
```

Comment:
Possibly some missing `if (field == NULL)`... which requires making the function return `int` instead of `void` and handle the error everywhere.